service/src/main/java/bio/terra/workspace/service/iam/model/WsmIamRole.java

```diff
@@ -73,6 +73,22 @@ public static Optional<WsmIamRole> getHighestRole(UUID workspaceId, List<WsmIamR
         String.format("Workspace %s has unexpected roles: %s", workspaceId.toString(), roles));
   }
 
+  public static boolean roleAtLeastAsHighAs(WsmIamRole minimumRole, WsmIamRole roleToCheck) {
```

Comment:
I suggest instead of doing this as a `switch`, you put the logic into the enumeration itself.
You could either put a comparison lambda directly in the enum or put an array of roles into the enum and test it in this method.

Reply:
I tried putting lambda in enum. Java said `Illegal self reference`:

![image](https://user-images.githubusercontent.com/10929390/183975261-28417598-116f-4a16-a234-b92ca1d04636.png)

Then I tried storing `Set<WsmIamRole> higherRoles` on enum. Java said `Illegal forward reference`.

![image](https://user-images.githubusercontent.com/10929390/183975358-9dc885a3-e854-4e0b-9e3d-be62b16d4090.png)

Reply:
TIL...